light-blockchain/src/push.rs

```diff
@@ -105,21 +100,27 @@ impl LightBlockchain {
         let chain_info = ChainInfo::from_block(block, &prev_info);
 
         // More chain ordering.
-        match chain_order {
+        let result = match chain_order {
             ChainOrdering::Extend => {
                 return LightBlockchain::extend(this, chain_info, prev_info);
             }
             ChainOrdering::Superior => {
                 return LightBlockchain::rebranch(this, chain_info, prev_info);
             }
-            ChainOrdering::Inferior => unreachable!(),
-            ChainOrdering::Unknown => {}
-        }
+            ChainOrdering::Inferior => {
+                log::debug!(block = %chain_info.head, "Storing block - on inferior chain");
+                PushResult::Ignored
+            }
+            ChainOrdering::Unknown => {
+                log::debug!(block = %chain_info.head, "Storing block - on fork");
```

Comment:
Seems strange that we're logging this here and not in the other cases. Also I'd change the message to something like:
`log::debug!(block = %chain_info.head, "Forked block - on fork")`

Reply:
same as above